src/_P122_SHT2x.ino

```diff
@@ -0,0 +1,265 @@
+#include "_Plugin_Helper.h"
+#ifdef USES_P122
+
+// #######################################################################################################
+// ######################## Plugin 122 SHT2x I2C Temperature Humidity Sensor  ############################
+// #######################################################################################################
+// 26-03-2023 Flashmark creation based upon https://github.com/RobTillaart/SHT2x
+
+# include "src/PluginStructs/P122_data_struct.h"
+
+#define PLUGIN_122
+#define PLUGIN_ID_122     122               // plugin id
+#define PLUGIN_NAME_122   "Environment - SHT2x" // What will be dislpayed in the selection list
+#define PLUGIN_VALUENAME1_122 "Temperature" // variable output of the plugin. The label is in quotation marks
+#define PLUGIN_VALUENAME2_122 "Humidity"    // multiple outputs are supported
+
+//   PIN/port configuration is stored in the following:
+//   CONFIG_PIN1 - The first GPIO pin selected within the task
+//   CONFIG_PIN2 - The second GPIO pin selected within the task
+//   CONFIG_PIN3 - The third GPIO pin selected within the task
+//   CONFIG_PORT - The port in case the device has multiple in/out pins
+//
+//   Custom configuration is stored in the following:
+//   PCONFIG(x)
+//   x can be between 1 - 8 and can store values between -32767 - 32768 (16 bit)
+//
+//   N.B. these are aliases for a longer less readable amount of code. See _Plugin_Helper.h
+//
+//   PCONFIG_LABEL(x) is a function to generate a unique label used as HTML id to be able to match 
+//                    returned values when saving a configuration.
+
+// Make accessing specific parameters more readable in the code
+// #define Pxxx_OUTPUT_TYPE_INDEX  2
+# define P122_I2C_ADDRESS         PCONFIG(0)
+# define P122_I2C_ADDRESS_LABEL   PCONFIG_LABEL(0)
+# define P122_RESOLUTION          PCONFIG(1)
+# define P122_RESOLUTION_LABEL    PCONFIG_LABEL(1)
+
+// A plugin has to implement the following function
+
+boolean Plugin_122(uint8_t function, struct EventStruct *event, String& string)
+{
+  // function: reason the plugin was called
+  // event: ??add description here??
+  // string: ??add description here??
+
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      // This case defines the device characteristics
+      Device[++deviceCount].Number           = PLUGIN_ID_122;
+      Device[deviceCount].Type               = DEVICE_TYPE_I2C;
+      Device[deviceCount].VType              = Sensor_VType::SENSOR_TYPE_TEMP_HUM;
+      Device[deviceCount].Ports              = 0;
+      Device[deviceCount].PullUpOption       = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption      = true;
+      Device[deviceCount].ValueCount         = 2;
+      Device[deviceCount].SendDataOption     = true;
+      Device[deviceCount].TimerOption        = true;
+      Device[deviceCount].I2CNoDeviceCheck   = true;
```

Comment:
Hm, doesn't the sensor handle the I2C protocol 'alive' check without issues, so this needs to be set? 🤔

Reply:
I copied this setting from P014, no clue what it does. Is there some documentation on the plugin "environment" telling what each of these setting mean and how the framework handles each of the functions in the plugin?
Seems it disables some kind of check, let's see if we can do without.

Reply:
That specific I2C setting can only be verified by actual testing of the sensor for some time, and could be added if it _isn't_ working properly, now that we've added a test at initialization of the plugin on PLUGIN_INIT (single failure), and for each PLUGIN_READ (10 consecutive read-failures) will disable the plugin.
So if the plugin returns invalid results after some time being active (hours to days), or returning some irregular values from time to time, that's a sign that the sensor doesn't handle these checks well, and only then this setting should be added.